### PR TITLE
Ensure to use valid MapRouteArrow and MapRouteLine Layer references during style change

### PR DESCRIPTION
**Pull request by LukasPaczos**
## Description
Closes https://github.com/mapbox/mapbox-navigation-android/issues/2305.
The `MapRouteArrow` has been misusing the Maps SDK by caching the `Layer`s and trying to access their values after the style has started to change. Previously, Maps SDK was removing the layers when style started to change, but that's not the case anymore to avoid blinking of resources, which exposed the issue.

I also discovered that `MapRouteArrow` is not re-appearing after style change - https://github.com/mapbox/mapbox-navigation-android/issues/2314. This is also reproducible on `master` before any changes from https://github.com/mapbox/mapbox-navigation-android/pull/2262.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Implementation

Instead of caching layer instances, I'm caching IDs instead. Then, I can safely use those IDs and verify if the layer and style are present.

### Testing
@Guardiola31337 is there an integration test that can be reused? It's surprising that this hasn't been caught before.

### CODE REVIEW

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteArrow.java [hunk 1]
```diff
@@ -85,7 +86,7 @@
   @ColorInt
   private final int arrowBorderColor;
 
-  private List<Layer> arrowLayers;
+  private List<String> arrowLayers;
```
Comment (Lebedinsky) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteArrow.java]: Should we rename it to `arrowLayersIds` ? Or smth like that

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteArrow.java [hunk 2]
```diff
@@ -123,10 +124,16 @@ void addUpcomingManeuverArrow(RouteProgress routeProgress) {
   }
 
   void updateVisibilityTo(boolean visible) {
-    for (Layer layer : arrowLayers) {
-      String targetVisibility = visible ? VISIBLE : NONE;
-      if (!targetVisibility.equals(layer.getVisibility().getValue())) {
-        layer.setProperties(visibility(targetVisibility));
+    Style style = mapboxMap.getStyle();
+    if (style != null) {
```
Comment (Lebedinsky) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteArrow.java]:
```suggestion
    mapboxMap.getStyle()?.let { style ->
```
wdyt?

Comment (LukasPaczos) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteArrow.java]: I deliberately used the null-checked one. We don't want to wait for style to be available - if it's not there, we should just skip. The upstream logic takes care of reloading the class when a new style is available.

Comment (LukasPaczos) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteArrow.java]: Oh, you suggested `let`. This is a java class :)

Comment (Lebedinsky) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteArrow.java]: Oh, sorry, my fault, it is Java :)

**Comment by Guardiola31337** [comment]
@LukasPaczos 
> I also discovered that `MapRouteArrow` is not re-appearing after style change - https://github.com/mapbox/mapbox-navigation-android/issues/2314. This is also reproducible on `master` before any changes from https://github.com/mapbox/mapbox-navigation-android/pull/2262.

Great catch 👍 

> is there an integration test that can be reused? It's surprising that this hasn't been caught before.

Unfortunately not. In terms of testing, the UI SDK needs definitely some ❤️ Feel free to take a stab at it, that'd be 💯 One thing that we could do though (relatively faster) is including an example (diff from https://github.com/mapbox/mapbox-navigation-android/issues/2305#issuecomment-565165796) in the test app as we have a couple of checks in the PR template testing section to 👀 ✅ before merging a PR

> - [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
> - [ ] I have tested via a test drive, or a simulation/mock location app

Although, as I said, I completely agree and we should start adding UI tests starting with these regressions we're finding.

**Comment by LukasPaczos** [comment]
https://github.com/mapbox/mapbox-navigation-android/pull/2315/commits/ceb3fbbb0d5bb14c099175c0feaee508459cb003 adds the regression example for future QA.

**Comment by LukasPaczos** [comment]
With the above example, I was able to reproduce the same issue for `MapRouteLine`:
```
A/DEBUG: backtrace:
A/DEBUG:       #00 pc 00000000000c3e84  /data/app/com.mapbox.services.android.navigation.app-wO5SYBaovGnSyT0uQM_EmQ==/lib/arm64/libmapbox-gl.so (BuildId: 8aaa709acceed6c52c4216aa9ff9fd17b78c0a84)
A/DEBUG:       #01 pc 00000000021001bc  /memfd:/jit-cache (deleted) (art_jni_trampoline+124)
A/DEBUG:       #02 pc 000000000203c9ec  /memfd:/jit-cache (deleted) (com.mapbox.mapboxsdk.style.layers.Layer.getId+60)
A/DEBUG:       #03 pc 00000000021037f8  /memfd:/jit-cache (deleted) (com.mapbox.services.android.navigation.ui.v5.route.MapRouteLine.updateAlternativeVisibilityTo+344)
```

This class, similarly to `MapRouteArrow` shouldn't store raw references to layers (because of how logic that accesses them is executed). I can fix it as part of this PR.

**Comment by LukasPaczos** [comment]
Above is fixed in https://github.com/mapbox/mapbox-navigation-android/pull/2315/commits/8d21e5c98a6e5d086ef54e81feada7dd477932c3. Can I get another quick round of reviews?

**Comment by Guardiola31337** [comment]
> - [x] Cherry-pick into [`base-v0.42.1`](https://github.com/mapbox/mapbox-navigation-android/tree/base-v0.42.1) branch

Fixed in https://github.com/mapbox/mapbox-navigation-android/pull/2318

**Pull request merged**